chaosgcp/apphub/actions.py

```diff
@@ -0,0 +1,315 @@
+        region (str): The region of the AppHub application.
+        target_name (str): the the name of a path matcher in the URL map.
+        target_path (str): default "/*",
+        impacted_percentage: float default 50.0,
+        http_status (int) default 400,
+        regional: (bool) default True,
+        configuration: (Configuration) default None,
+        secrets: (Secrets) default None,
+    """
+
+    if url_map_exists(
+        url_map,
+        host_project_id,
+        project_id,
+        region,
+        application_name,
+        configuration,
+        secrets,
+    ):
+        # URL map exists, proceed with your processing logic here
+        logger.info(
+            f"Given URL map '{url_map}' found in given Apphub Application. Proceeding with Fault Injection Action "
+        )
+        inject_traffic_faults(
+            url_map,
+            target_name,
+            target_path,
+            impacted_percentage,
+            http_status,
+            regional,
+            project_id,
+            region,
+            configuration,
+            secrets,
+        )
+    else:
+        raise ActivityFailed(
+            f"Given URL map '{url_map}' not found in given Apphub Application, Cannot Inject Fault "
+        )
+
+
+def remove_fault_if_url_map_exists_app_hub(
+    application_name: str,
+    host_project_id: str,
+    url_map: str,
+    project_id: str,
+    region: str,
+    target_name: str,
+    target_path: str = "/*",
+    regional: bool = True,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+) -> None:
+    """Processes a URL map if it exists within the specified AppHub application.
+
+    Args:
+        application_name (str): The name of the AppHub application.
+        host_project_id (str): The ID of the Host Apphub project.
+        url_map (str): The URL map value to check for.
+        project_id (str): The Google Cloud project ID.
+        region (str): The region of the AppHub application.
+        target_name (str): the the name of a path matcher in the URL map.
+        target_path (str): default "/*",
+        regional: (bool) default True,
+        configuration: (Configuration) default None,
+        secrets: (Secrets) default None,
+    """
+
+    if url_map_exists(
+        url_map,
+        host_project_id,
+        project_id,
+        region,
+        application_name,
+        configuration,
+        secrets,
+    ):
+        # URL map exists, proceed with your processing logic here
+        logger.info(
+            f"URL map '{url_map}' - Rollback of Fault Injection Action "
+        )
+        remove_fault_injection_traffic_policy(
+            url_map,
+            target_name,
+            target_path,
+            regional,
+            project_id,
+            region,
+            configuration,
+            secrets,
+        )
+    else:
+        raise ActivityFailed(
+            f"Given URL map '{url_map}' not found, RollBack not required "
+        )
```

Comment:
for a cleaner approach IMO, try this:

```python
if not url_map_exists(
        url_map,
        host_project_id,
        project_id,
        region,
        application_name,
        configuration,
        secrets,
    ):
     raise ActivityFailed(
            f"Given URL map '{url_map}' not found, RollBack not required "
        )

# URL map exists, proceed with your processing logic here
logger.info(
    f"URL map '{url_map}' - Rollback of Fault Injection Action "
)
remove_fault_injection_traffic_policy(
    url_map,
    target_name,
    target_path,
    regional,
    project_id,
    region,
    configuration,
    secrets,
)    
```

So that you have a short `if` block and the main code level remains outside of the conditionnal.

Reply:
changed .